include/dlaf/mc/cholesky_local.h

```diff
@@ -0,0 +1,90 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#include <hpx/include/parallel_executors.hpp>
+
+#include "dlaf/blas_tile.h"
+#include "dlaf/lapack_tile.h"
+#include "dlaf/matrix.h"
+#include "dlaf/util_matrix.h"
+
+/// @file
+
+namespace dlaf {
+
+/// @brief Cholesky implementation on local memory
+///
+/// @param uplo specifies that the matrix is \a Lower triangular
+/// @tparam mat refers to a dlaf::Matrix object
+///
+/// @throws std::runtime_error if \p uplo = \a Upper decomposition is chosen (not yet implemented)
+
+template <class T>
+void cholesky_local(blas::Uplo uplo, Matrix<T, Device::CPU>& mat) {
+  // Set up executor on the default queue with high priority.
+  hpx::threads::scheduled_executor matrix_HP_executor =
+      hpx::threads::executors::pool_executor("default", hpx::threads::thread_priority_high);
+
+  // Set up executor on the default queue with default priority.
+  hpx::threads::scheduled_executor matrix_normal_executor =
+      hpx::threads::executors::pool_executor("default", hpx::threads::thread_priority_default);
+
+  // Check if matrix is square
+  util_matrix::assert_size_square(mat, "Cholesky", "mat");
+  // Check if block matrix is square
+  util_matrix::assert_blocksize_square(mat, "Cholesky", "mat");
+
+  // Number of tile (rows = cols)
+  SizeType nrtile = mat.nrTiles().cols();
+
+  if (uplo == blas::Uplo::Lower) {
+    for (SizeType k = 0; k < nrtile; ++k) {
+      // Cholesky decomposition on mat(k,k) r/w potrf (lapack operation)
+
+      hpx::dataflow(matrix_HP_executor, hpx::util::unwrapping(tile::potrf<T, Device::CPU>), uplo,
+                    std::move(mat({k, k})));
+
+      for (SizeType i = k + 1; i < nrtile; ++i) {
+        // Update panel mat(i,k) with trsm (blas operation), using data mat.read(k,k)
+        hpx::dataflow(matrix_HP_executor, hpx::util::unwrapping(tile::trsm<T, Device::CPU>),
+                      blas::Side::Right, uplo, blas::Op::ConjTrans, blas::Diag::NonUnit, 1.0,
+                      mat.read({k, k}), std::move(mat({i, k})));
+      }
+
+      for (SizeType j = k + 1; j < nrtile; ++j) {
+        // Choose queue priority
+        auto trailing_matrix_executor = matrix_normal_executor;
+        if (j == k + 1) {
+          trailing_matrix_executor = matrix_HP_executor;
+        }
+        else {
+          trailing_matrix_executor = matrix_normal_executor;
+        }
```

Comment:
Am I wrong saying that the `else` branch is not very useful?

Moreover, it seems that the priority is set differently just for the first iteration, maybe this code can be changed to not use the `if`. Anyway, it is not a performance problem here.